compile error

Compiling ort v2.0.0-rc.9

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtLoraAdapter)` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtLoraAdapter)`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/adapter.rs:26:3
    |
26  |         ortsys![unsafe ReleaseLoraAdapter(self.ptr.as_ptr())];
    |         ----------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)
error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtValue, *mut *mut c_void) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:261:66
    |
261 | ...ate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) })?;
    |                       ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtValue, *mut *mut c_void) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/impl_tensor/mod.rs:91:3
    |
91  |         ortsys![unsafe GetTensorMutableData(self.ptr_mut(), &mut buffer_ptr)?; nonNull(buffer_ptr)];
    |         ------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtValue, *mut *mut c_void) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:261:66
    |
261 | ...ate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) })?;
    |                       ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtValue, *mut *mut c_void) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/impl_tensor/mod.rs:114:3
    |
114 |         ortsys![unsafe GetTensorMutableData(self.ptr().cast_mut(), &mut buffer_ptr)?; nonNull(buffer_ptr)];
    |         -------------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtValue, *const i64, usize, *mut *mut c_void) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:239:66
    |
239 | ...).$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }).expect($e)
    |              ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtValue, *const i64, usize, *mut *mut c_void) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/impl_tensor/mod.rs:240:3
    |
240 |         ortsys![unsafe TensorAt(self.ptr().cast_mut(), index.as_ptr(), N, &mut out).expect("Failed to index tensor")];
    |         ------------------------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtValue, *const i64, usize, *mut *mut c_void) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:239:66
    |
239 | ...).$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }).expect($e)
    |              ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtValue, *const i64, usize, *mut *mut c_void) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/impl_tensor/mod.rs:251:3
    |
251 |         ortsys![unsafe TensorAt(self.ptr_mut(), index.as_ptr(), N, &mut out).expect("Failed to index tensor")];
    |         ------------------------------------------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut ONNXType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut ONNXType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:83:3
    |
83  |         ortsys![unsafe GetOnnxTypeFromTypeInfo(typeinfo_ptr, &mut ty)]; // infallible
    |         -------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:87:5
    |
87  |                 ortsys![unsafe CastTypeInfoToTensorInfo(typeinfo_ptr, &mut info_ptr)]; // infallible
    |                 --------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtSequenceTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtSequenceTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:92:5
    |
92  |                 ortsys![unsafe CastTypeInfoToSequenceTypeInfo(typeinfo_ptr, &mut info_ptr)]; // infallible
    |                 --------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtSequenceTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtSequenceTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:95:5
    |
95  |                 ortsys![unsafe GetSequenceElementType(info_ptr, &mut element_type_info)]; // infallible
    |                 ------------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut ONNXType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut ONNXType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:98:5
    |
98  |                 ortsys![unsafe GetOnnxTypeFromTypeInfo(element_type_info, &mut ty)]; // infallible
    |                 ------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:103:7
    |
103 |                         ortsys![unsafe CastTypeInfoToTensorInfo(element_type_info, &mut info_ptr)]; // infallible
    |                         -------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtMapTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtMapTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:109:7
    |
109 |                         ortsys![unsafe CastTypeInfoToMapTypeInfo(element_type_info, &mut info_ptr)]; // infallible
    |                         --------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtMapTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtMapTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:118:5
    |
118 |                 ortsys![unsafe CastTypeInfoToMapTypeInfo(typeinfo_ptr, &mut info_ptr)]; // infallible
    |                 ---------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtOptionalTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtOptionalTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:123:5
    |
123 |                 ortsys![unsafe CastTypeInfoToOptionalTypeInfo(typeinfo_ptr, &mut info_ptr)]; // infallible
    |                 --------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtOptionalTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtOptionalTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:126:5
    |
126 |                 ortsys![unsafe GetOptionalContainedTypeInfo(info_ptr, &mut contained_type)]; // infallible
    |                 --------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtTypeInfo)` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtTypeInfo)`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:132:3
    |
132 |         ortsys![unsafe ReleaseTypeInfo(typeinfo_ptr)];
    |         --------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut *mut OrtTensorTypeAndShapeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut *mut OrtTensorTypeAndShapeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:140:5
    |
140 |                 ortsys![unsafe CreateTensorTypeAndShapeInfo(&mut info_ptr)];
    |                 ----------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, ONNXTensorElementDataType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, ONNXTensorElementDataType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:141:5
    |
141 |                 ortsys![unsafe SetTensorElementType(info_ptr, (*ty).into())];
    |                 ------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, *const i64, usize) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, *const i64, usize) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:142:5
    |
142 |                 ortsys![unsafe SetDimensions(info_ptr, dimensions.as_ptr(), dimensions.len())];
    |                 ------------------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, *mut *const i8, usize) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtTensorTypeAndShapeInfo, *mut *const i8, usize) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:149:5
    |
149 |                 ortsys![unsafe SetSymbolicDimensions(info_ptr, dimension_symbols.as_ptr().cast_mut(), dimension_symbols.len())];
    |                 --------------------------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:247:2
    |
247 |     ortsys![GetTensorElementType(info_ptr, &mut type_sys)];
    |     ------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut usize) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut usize) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:251:2
    |
251 |     ortsys![GetDimensionsCount(info_ptr, &mut num_dims)];
    |     ---------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut i64, usize) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut i64, usize) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:254:2
    |
254 |     ortsys![GetDimensions(info_ptr, node_dims.as_mut_ptr(), num_dims)];
    |     ------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut *const i8, usize) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut *const i8, usize) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:257:2
    |
257 |     ortsys![GetSymbolicDimensions(info_ptr, symbolic_dims.as_mut_ptr(), num_dims)];
    |     ------------------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtMapTypeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtMapTypeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:273:2
    |
273 |     ortsys![GetMapKeyType(info_ptr, &mut key_type_sys)]; // infallible
    |     --------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtMapTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtMapTypeInfo, *mut *mut OrtTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:277:2
    |
277 |     ortsys![GetMapValueType(info_ptr, &mut value_type_info)]; // infallible
    |     -------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTypeInfo, *mut *const OrtTensorTypeAndShapeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:279:2
    |
279 |     ortsys![unsafe CastTypeInfoToTensorInfo(value_type_info, &mut value_info_ptr)]; // infallible
    |     ------------------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:230:25
    |
230 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+)
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtTensorTypeAndShapeInfo, *mut ONNXTensorElementDataType) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/type.rs:281:2
    |
281 |     ortsys![GetTensorElementType(value_info_ptr, &mut value_type_sys)]; // infallible
    |     ------------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*mut OrtValue)` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*mut OrtValue)`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/mod.rs:72:4
    |
72  |             ortsys![unsafe ReleaseValue(ptr)];
    |             --------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtValue, *mut *mut OrtTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtValue, *mut *mut OrtTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/mod.rs:300:3
    |
300 |         ortsys![unsafe GetTypeInfo(ptr.as_ptr(), &mut typeinfo_ptr)];
    |         ------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtValue, *mut *mut OrtTypeInfo) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtValue, *mut *mut OrtTypeInfo) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/mod.rs:318:3
    |
318 |         ortsys![unsafe GetTypeInfo(ptr.as_ptr(), &mut typeinfo_ptr)];
    |         ------------------------------------------------------------ in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn(*const OrtValue, *mut i32) -> OrtStatusPtr` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:233:34
    |
233 |         unsafe { $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))($($n),+) }
    |                                        ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn(*const OrtValue, *mut i32) -> OrtStatusPtr`
    |
   ::: /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/value/mod.rs:368:3
    |
368 |         ortsys![unsafe IsTensor(self.ptr(), &mut result)]; // infallible
    |         ------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `unwrap_or_else` found for fn pointer `unsafe extern "system" fn() -> *const i8` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:227:25
    |
121 |     let str = unsafe { ortsys![GetBuildInfoString]() };
    |                        --------------------------- in this macro invocation
...
227 |         $crate::api().$method.unwrap_or_else(|| unreachable!(concat!("Method `", stringify!($method), "` is null")))
    |                               ^^^^^^^^^^^^^^ method not found in `unsafe extern "system" fn() -> *const i8`
    |
    = note: this error originates in the macro `ortsys` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `expect` found for fn pointer `unsafe extern "system" fn(u32) -> *const OrtApi` in the current scope
   --> /Users/rey/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ort-2.0.0-rc.9/src/lib.rs:200:21
    |
200 |                     (*base).GetApi.expect("`GetApi` must be present in `OrtApiBase`");
    |                                    ^^^^^^ method not found in `unsafe extern "system" fn(u32) -> *const OrtApi`

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `ort` (lib) due to 233 previous errors

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,6 +32,7 @@ ort = { version = "=2.0.0-rc.9", default-features = false, features = [
 rayon = { version = "1.10", default-features = false }
 serde_json = { version = "1" }
 tokenizers = { version = "0.21", default-features = false, features = ["onig"] }
+ort-sys = { version = "=2.0.0-rc.9", default-features = false }
 
 [features]
 default = ["ort-download-binaries", "hf-hub-native-tls"]

